types/mempool/sender_nonce.go

```diff
@@ -0,0 +1,207 @@
+package mempool
+
+import (
+	crand "crypto/rand" // #nosec // crypto/rand is used for seed generation
+	"encoding/binary"
+	"fmt"
+	"math/rand" // #nosec // math/rand is used for random selection and seeded from crypto/rand
+
+	huandu "github.com/huandu/skiplist"
```

Comment:
Could we change this? I don't like having stuff like `orderedSenders := huandu.New(huandu.String)`; `skiplist.New()` is better

```suggestion
	skiplist "github.com/huandu/skiplist"
```